legion/legion/utils.py

```diff
@@ -588,3 +591,30 @@ def deduce_model_file_name(model_id, model_version):
         return os.path.join(default_prefix, file_name)
 
     return file_name
+
+
+def retry_function_call(function_to_call, retries, timeout):
+    """
+    Try to call function till it will return not None object.
+    Raise if there are no retries left
+
+    :param function_to_call: function to be called
+    :type function_to_call: Callable[[], any]
+    :param retries: count of retries
+    :type retries: int
+    :param timeout: timeout between retries
+    :type timeout: int
+    :return: Any -- result of successful function call
+    """
+    for no in range(retries):
+        result = function_to_call()
+        if result is not None:
+            return result
+
+        if no < retries - 1:
+            LOGGER.debug('Retry {}/{} was failed. Waiting {}s before next retry analysis'.format(no + 1,
+                                                                                                 retries,
+                                                                                                 timeout))
+            time.sleep(timeout)
+
+    raise Exception('No retries left')
```

Comment:
I'd rather return False here in case if the internal function does not succed. Then you can handle the error on a caller's side  an provide more meaningful error message.

Reply:
Okay, will do